src/Microsoft.AspNetCore.Hosting/GenericHost/GenericWebHostedService.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting.Builder;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Hosting.Views;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.StackTrace.Sources;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    internal class GenericWebHostService : IHostedService
+    {
+        private static readonly string DeprecatedServerUrlsKey = "server.urls";
+
+        public GenericWebHostService(IOptions<GenericWebHostServiceOptions> options,
```

Comment:
Something interesting here, we should maybe delay resolution of these services until StartAsync. If any of these things fail to resolve, you won't get startup errors in the browser.

Reply:
What of these things aren't registered by hosting?

Reply:
Wait, I just realized, if the server can't be resolved we can't capture startup errors and do anything meaningful anyways 😄 . This is fine.